### ISSUE TITLE
Re-run the script that adds the string sort attribute

### ISSUE THREAD

**Issue by wlpotter** [srophe/caesarea-data#177]
Now that we have new data we need to update so that the records ahve the string sort attribute letting records within the same work appear properly sorted w.r.t. the section ordering.

**Add a test to this script so that it only runs on work-groups that have at least one record that _lacks_ the n attribute.** This will cut down on time re-processing unchanged work groups and, more importantly the time spent manually checking the ones that couldn't be sorted by the matching algorithm.

**Comment by wlpotter** [comment]
Hmm I might have uncovered a second problem:

We have several cases of the same work being referred to separately. And since we don't have stable URIs for all authors and works, I have had to rely on string matching to create the author-work groups for sorting. Here are the offenders as of 2023-02-24:

Eusebius of Caesarea. Dictionary of Place Names
Eusebius of Caesarea. Ecclesiastical History
Eusebius of Caesarea. Historia ecclesiastica
Eusebius of Caesarea. Life of Constantine
Eusebius of Caesarea. Martyrs of Palestine
Eusebius. Dictionary of Place Names
Eusebius. Life of Constantine
Photius. Library
Photius. Library Library
Pliny the Elder. Natural History
Pliny. Natural History
Procopius of Gaza. Letter
Procopius of Gaza. Letters
Socrates of Constantinople. Church History
Socrates Scholasticus. Ecclesiastical History
Sozomen. Church History
Sozomenus. Church History


We will need some data normalization before we can reliably run an updated string sort. @davidamichelson we should discuss.

The string sort is not essential, and ideally we'd just run it once more before official publication, so this is not a priority.

**Comment by davidamichelson** [comment]
@wlpotter here are the revisions for making these uniform.

Eusebius of Caesarea. Historia ecclesiastica -> Eusebius of Caesarea. Ecclesiastical History
Eusebius. Dictionary of Place Names -> Eusebius of Caesarea. Dictionary of Place Names
Eusebius. Life of Constantine -> Eusebius of Caesarea. Life of Constantine
Photius. Library Library -> Photius. Library
Pliny. Natural History -> Pliny the Elder. Natural History
Procopius of Gaza. Letters -> Procopius of Gaza. Letter
Socrates Scholasticus. Ecclesiastical History -> Socrates of Constantinople. Church History
Sozomen. Church History -> Sozomenus. Church History

**Comment by davidamichelson** [comment]
@wlpotter please make an issue to discuss "Letters vs. Letter" @josephrife on Friday please.

@wlpotter Also one more to discuss, should be add geographic names for other authors when there are multiple with the same name (Procopius. Secret History  -> Procopius of Caesarea -> Secret History)

**Comment by josephrife** [comment]
Thanks. On the second question with with e.g. Procopius of Caesarea. Good catch. An editing error I suspect, we just correct by adding "of Caesarea" to the other Procopius refs that are missing it. I am not sure if there are other examples, I might recall seeing some "Pliny the Elder" but then a bare "Pliny" that needs enhancement too. Possible to make a list of all "authors" and see where there are any to fix?


-----------------------------------------------------------------
Joseph L. Rife
Associate Professor
Department of Classical and Mediterranean Studies
Affiliated Faculty in Anthropology and Religion
Director, The American Excavations at Kenchreai (Greece)
Co-Director, The Caesarea City and Port Exploration Project (Israel)

Check out my research websites:
https://caesarea-maritima.org/
http://www.kenchreai.org/

Check out my books:
[cid:c21826f5-54ca-451a-9922-6a6163d57000]      [cid:7130c953-93e1-4d0e-8e97-dca894f9950e] <https://www.ascsa.edu.gr/publications/book/?i=9780876615546>




________________________________

**Comment by wlpotter** [comment]
Note to self:

- split this issue into one for normalizing titles and authors (the list in the above comment); one for creating the report to check all titles and authors; and then leave this for re-running the string sort

**Comment by wlpotter** [comment]
This should be fixed
